lib/fluent/plugin/out_google_cloud.rb

```diff
@@ -665,9 +688,22 @@ def write(chunk)
           rescue Google::Apis::ClientError => error
             # Most ClientErrors indicate a problem with the request itself and
             # should not be retried.
-            increment_dropped_entries_count(entries_count)
-            @log.warn "Dropping #{entries_count} log message(s)",
-                      error_class: error.class.to_s, error: error.to_s
+            error_details_map = {}
+            error_details_map = construct_error_details_map(error) if
```

Comment:
This can now become:
```ruby
error_details_map = if @partial_success then construct_error_details_map(error) else {} end
```
Or you could even add `return {} unless @partial_success` at the start of `construct_error_details_map` and make this an unconditional assignment.

Reply:
Changed.